### ISSUE TITLE
CMake doesn't find /lib/x86_64-linux-gnu/libOpenCL.so.1 to enable OpenCL

### ISSUE THREAD

**Issue by pjaaskel** [CHIP-SPV/chipStar#542]
This happens in the Intel Devcloud PVC env (at least). It's in ld.so.conf and clinfo works.

**Comment by pjaaskel** [comment]
Added workaround to https://github.com/CHIP-SPV/chipStar/blob/Release-1.0/README.md, not blocking 1.0.

**Comment by pjaaskel** [comment]
Added workaround to https://github.com/CHIP-SPV/chipStar/blob/Release-1.0/README.md, not blocking 1.0.

**Comment by pjaaskel** [comment]
Seems CMake searches only for *.so. In our case since we ship OpenCL headers, we could just use the so.1 without installing ocl-icd-opencl-dev which brings in the .so. An ugly workaround:

```
mkdir -p ~/local/lib
ln -s /usr/lib/x86_64-linux-gnu/libOpenCL.so.1 ~/local/lib/libOpenCL.so
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$HOME/local/lib
cmake [...]
```

**Comment by franz** [comment]
`LD_LIBRARY_PATH` is only used when loading a library (by `ld`); it's not used when GCC or other compiler searches for a library at build time. CMake has its [own variable](https://cmake.org/cmake/help/latest/variable/CMAKE_LIBRARY_PATH.html) for this purpose.

**Comment by pjaaskel** [comment]
Yeah, so I thought before taking a look at chipStar's CMake conf which uses it directly.

**Comment by pvelesko** [comment]
I did that on purpose because I was getting tired of default OpenCL runtimes getting picked up. Normally I log in, load modules (which load LD_LIBRARY_PATH) and then compile and run.

The fix is as simple as enabling searches on default path.

**Comment by pjaaskel** [comment]
Added a workaround to README.md. Should do for 1.0.

**Comment by pvelesko** [comment]
Is this a name issue? Cmake expectes libOpenCL.so and libOpenCL.so.1 doens't satisfy it?

**Comment by pjaaskel** [comment]
I think so, yes.
